### PR TITLE
Route error codes

### PR DESCRIPTION
**Pull request by camotts**

### CODE REVIEW

File: router/xgress_edge_tunnel/dialer.go [hunk 1]
```diff
@@ -51,7 +52,7 @@ func (self *tunneler) Dial(destination string, circuitId *identity.TokenId, addr
```
Comment (plorenz) [inline router/xgress_edge_tunnel/dialer.go:52]: Above, where `self.terminators.get(destination)` returns an error if the terminator is not found is where we'd want to return an InvalidTerminator error

File: router/xgress_edge/dialer.go [hunk 2]
```diff
@@ -124,7 +125,7 @@ func (dialer *dialer) Dial(destination string, circuitId *identity.TokenId, addr
 		if err != nil {
```
Comment (plorenz) [inline router/xgress_edge/dialer.go:125]: We'd also want to return an InvalidTerminator error above, if `listenConn, found := dialer.factory.hostedServices.Get(token)` doesn't find the terminator.

File: router/xgress_edge/dialer.go [hunk 3]
```diff
@@ -124,7 +125,7 @@ func (dialer *dialer) Dial(destination string, circuitId *identity.TokenId, addr
...
 			conn.close(false, err.Error())
 			x.Close()
```
Comment (plorenz) [inline router/xgress_edge/dialer.go:127]: This one should just be a regular dial error. Terminator exists, but had trouble dialing.

File: router/xgress_edge_transport/dialer.go [hunk 4]
```diff
@@ -52,13 +54,13 @@ func (txd *dialer) Dial(destination string, circuitId *identity.TokenId, address
 
 	txDestination, err := transport.ParseAddress(destination)
 	if err != nil {
-		return nil, fmt.Errorf("cannot dial on invalid address [%s] (%s)", destination, err)
+		return nil, xgress.InvalidTerminatorError{InnerError: fmt.Errorf("cannot dial on invalid address [%s] (%s)", destination, err)}
 	}
 
 	log.Debug("dialing")
 	peer, err := txDestination.Dial("x/"+circuitId.Token, circuitId, txd.options.ConnectTimeout, nil)
 	if err != nil {
```
Comment (plorenz) [inline router/xgress_edge_transport/dialer.go:62]: Should be a regular dial error

File: router/xgress_edge_tunnel/dialer.go [hunk 5]
```diff
@@ -51,7 +52,7 @@ func (self *tunneler) Dial(destination string, circuitId *identity.TokenId, addr
 
 	conn, halfClose, err := terminator.context.Dial(options)
 	if err != nil {
```
Comment (plorenz) [inline router/xgress_edge_tunnel/dialer.go:54]: Should be a regular dial error

File: router/xgress_edge_transport/dialer.go (hunk 4 above)
Comment (camotts) [reply router/xgress_edge_transport/dialer.go:62]: This is how I am checking back in the controller if its an "expected" error. The dial error gets persisted in the inner error and checked on the controller side. Does that need to be redone?

**Review by plorenz — APPROVED** [review]

**Pull request merged**